rfcs/0000-ignore-type-info.md

```diff
@@ -0,0 +1,82 @@
+- Feature Name: ignore-type-info
+- Start Date: 2018-5-1
+- RFC PR: (leave this empty)
+- Feature Commit(s): (leave this empty)
+
+# Summary
+
+Add a syntax property that causes typechecking to ignore the synthesized and
+expected types and instead produce the type `Any` along with trivial
+propositions and objects. The expression must typecheck at the type `Any`.
+
+The property is named `typed-racket:ignore-type-information`. 
+
+# Motivation
+
+This provides built-in support for a pattern where the programmer
+wants write a test that is always true (or false) but the typechecker
+can't tell that. Currently, it's easy to do this as follows:
+
+```
+(if (> (random 5) 6) 'never 'always)
+```
+
+However, it's problematic to rely on the type checker never
+understanding things in the future, and also we'd like to enable the
+bytecode compiler to optimize the test. 
+
+This was developed specifically to support this pattern in a revised
+version of the keyword argument protocol currently under development
+by @mflatt.
+
+# Guide-level explanation
+
+Any expression with the shape `(#%expression sub)` that has a true
+value for the syntax property `typed-racket:ignore-type-information`
+will have the type `Any`, and the type checker won't learn anything
+about the expression for use in refining other types.
+
+The expression `sub` must still type check, but can have any
+single-valued type.
+
+# Reference-level explanation
+
+Forms of the `(#%expression sub)` that has a true value for the syntax
+property `typed-racket:ignore-type-information` type check the
+expression `sub` with expected type `Any`. The resulting `tc-result`
+has type `Any` along with `true-prop` for both propositions and
+`empty-obj` for the object.
+
+
+
+# Drawbacks and Alternatives
+[drawbacks]: #drawbacks
+
+Adds complexity to the type checker.
+
+#### Alternative: use expected type
```

Comment:
I think it should be just a little more clear, e.g. "use the expected type (if one exists) to type check the marked expression instead of using `Any` for the expected type -- and in either case the overall result will still be `Any`".

Reply:
No, that's not quite the intent of the alternative -- it's to use the expected type in both cases.